client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/settings.js

```diff
@@ -56,6 +56,7 @@ export const CarrierAccountSettings = ( props ) => {
 	if ( isConnectionSuccess ) {
 		const url = new URL( window.location.href );
 		url.searchParams.delete( 'carrier' );
+		window.onbeforeunload = null;
```

Comment:
Change looks good 👍 

I'm noticing this is creating a side effect on `render`.

I think at some point it might be best maybe to react-ify the component with something like
```
const OnConnectionSuccess = ({isConnectionSuccess}) => {
	React.useEffect( () => {
		if ( !isConnectionSuccess ) {
			return;
		}

		window.onbeforeunload = null;
		const url = new URL( window.location.href );
		url.searchParams.delete( 'carrier' );
		window.location.href = url.href;
	}, [isConnectionSuccess] );

	return null;
}

...

<OnConnectionSuccess isConnectionSuccess={isConnectionSuccess} />
```

Or to put it in some hook inside `CarrierAccountSettings`

Reply:
I really like this idea. I'm going to merge this PR for now, but will make an enhancement issue for this.

Reply:
Sounds great!